warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -447,19 +459,28 @@ public IndexInfo intersect(IndexInfo o) {
      * @return
      */
     public IndexInfo intersect(IndexInfo o, List<JexlNode> delayedNodes, UidIntersector uidIntersector) {
-        if (isInfinite() && !o.isInfinite()) {
-            /*
-             * A) we are intersecting UNKNOWN AND small
-             */
-            if (o.onlyEvents())
-                return intersect(Math.max(count, o.count), o.uids(), o.getNode(), Lists.newArrayList(getNode()), delayedNodes);
+        
+        // infinite = (count == -1)
+        // onlyEvents = (count == uids.size())
+        
+        // if either side is only events and the other is infinite, roll to a shard range
+        if ((isInfinite() && o.onlyEvents()) || (o.isInfinite() && this.onlyEvents())) {
+            JexlNodeSet nodes = new JexlNodeSet();
```

Comment:
This block is the same as at the end of the method.  Why have a special return here?  Also can this case simply be if (isInfinite() || o.isInfinite()) and would be handled already by the logic below at lines 503-

Reply:
Done.